vllm/v1/core/sched/scheduler.py

```diff
@@ -257,6 +257,8 @@ def schedule(self) -> SchedulerOutput:
                             self.running,
                             key=lambda r: (r.priority, r.arrival_time),
                         )
+                        if self.running.index(preempted_req) <= req_index:
+                            req_index -= 1
                         self.running.remove(preempted_req)
```

Comment:
Can avoid scanning the list twice (actually 3 times with the `max`)
```suggestion
                        preempted_index = self.running.index(preempted_req)
                        if preempted_index <= req_index:
                            req_index -= 1
                        self.running.pop(preempted_index)
```